[INTERNAL] CONTRIBUTING.md: Add section about multi branch development (#919)
SAP/ui5-tooling: CONTRIBUTING.md

    - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
1. Once the change has been approved and merged, we will inform you in a comment.
1. Celebrate! 🎉

> ### Multi Branch Development

Reviewer: ```suggestion
### Multi-Branch Development
```

There are phases when UI5 Tooling has already started developing a new major versions of it's projects but does not has released the new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current and the new major version.

> 

Reviewer 2: I would really like a short example here. Something like "Current version : v3; New Major: v4".

Otherwise looks great!

Author: done. If you're fine with the current state. I will assign UA.

> There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't yet released a new major version. This situation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).

Reviewer: ```suggestion
There are phases when development of a new major version of UI5 Tooling has already started, but this new major version hasn't been released yet. This situation requires a special handling of pull requests / commits if the actual coding change should be both part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```

> There are phases when UI5 Tooling has already started developing a new major version of it's projects but does not has released the new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).

Reviewer 3: ```suggestion
There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't released a new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```

> 1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or `[INTERNAL]`

Reviewer: ```suggestion
1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or `[INTERNAL]`.
```

> 2. Rebase and merge the PR in the current version

Reviewer: ```suggestion
2. Rebase and merge the PR in the current version.
```

> 3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change appears in the changelog of the new major version

Reviewer: ```suggestion
3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change does not appear in the changelog of the new major version
```

Author: done

> 3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change does not appear in the changelog of the new major version

Reviewer: ```suggestion
3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this, you ensure that the coding change does not appear in the changelog of the new major version.
```
